libsel4vm/src/arch/arm/vgic/vgic_v2.c

```diff
@@ -58,9 +58,70 @@
 #include "gicv2.h"
 #include "vdist.h"
 
+#include <libfdt.h>
 
+static uintptr_t vcpu_addr;
+static uintptr_t cpu_addr;
+static uintptr_t dist_addr;
 static struct vgic_dist_device *vgic_dist;
 
+static int gic_get_registers_from_fdt(vm_t *vm, uintptr_t *dist, uintptr_t *cpu, uintptr_t *vcpu)
+{
+    const struct fdt_property *c;
+    int len;
+
+    int gic_offset = fdt_path_offset(vm->fdt_ori, vm->gic_path);
+    if (0 > gic_offset) {
+        ZF_LOGE("Failed to find gic offset from path: %s", vm->gic_path);
+        return -1;
+    }
+
+    int parent_offset = fdt_parent_offset(vm->fdt_ori, gic_offset);
+    if (0 > parent_offset) {
+        ZF_LOGE("Failed to find gic parent offset");
+        return -1;
+    }
+
+    int address_cells = fdt_address_cells(vm->fdt_ori, parent_offset);
+    int size_cells = fdt_address_cells(vm->fdt_ori, parent_offset);
+
+    if ((address_cells == 0) || (size_cells == 0)) {
+        ZF_LOGE("#address-cells or #size-cells are 0");
+        return -1;
+    }
+
+    c = fdt_get_property(vm->fdt_ori, gic_offset, "reg", &len);
+    if (!c) {
+        ZF_LOGE("Failed to find reg in gic node");
+        return -1;
+    }
+
+    /*
+     * 1st reg entry: Distributor
+     * 2nd reg entry: CPU Interface
+     * 3rd reg entry: GIC Virtual Interface Control Register
+     * 4th reg entry: GIC Virtual CPU Interface Register
+     */
+    uint64_t gic_regs[NUM_GIC_REGS];
+    int calculated_size = size_cells * address_cells * NUM_GIC_REGS * sizeof(uint32_t);
```

Comment:
```suggestion
    int calculated_size = (address_cells + size_cells)  * NUM_GIC_REGS * sizeof(uint32_t);
```
We have one address cell followed by a size cell for each element. This just work because `2 * 2` happens to equal `2 + 2`